restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/GeolocationEndpoint.java

```diff
@@ -0,0 +1,732 @@
+                            data.putSingle("LocationAreaCode", token);
+                        }
+                        if (item.contains("cellid")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("CellId", token);
+                        }
+                        if (item.contains("aol")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("LocationAge", token);
+                        }
+                        if (item.contains("vlrNumber")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("NetworkEntityAddress", token);
+                        }
+                        if (item.contains("latitude")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("DeviceLatitude", token);
+                        }
+                        if (item.contains("longitude")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("DeviceLongitude", token);
+                        }
+                        if (item.contains("civicAddress")) {
+                            String token = item.substring(item.lastIndexOf("=") + 1);
+                            data.putSingle("FormattedAddress", token);
+                        }
+                    }
+                }
+                if (gmlcURI != null && gmlcResponse != null) {
+                    // For debugging/logging purposes only
+                    logger.info("Geolocation data of " + targetMSISDN + " retrieved from GMCL at: " + gmlcURI);
+                    logger.info("MCC (Mobile Country Code) = " + getInteger("MobileCountryCode", data));
+                    logger.info("MNC (Mobile Network Code) = " + data.getFirst("MobileNetworkCode"));
+                    logger.info("LAC (Location Area Code) = " + data.getFirst("LocationAreaCode"));
+                    logger.info("CI (Cell ID) = " + data.getFirst("CellId"));
+                    logger.info("AOL (Age of Location) = " + getInteger("LocationAge", data));
+                    logger.info("NNN (Network Node Number/Address) = " + +getLong("NetworkEntityAddress", data));
+                    logger.info("Devide Latitude = " + data.getFirst("DeviceLatitude"));
+                    logger.info("Devide Longitude = " + data.getFirst("DeviceLongitude"));
+                    logger.info("Civic Address = " + data.getFirst("FormattedAddress"));
+                }
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        Geolocation geolocation = createFrom(new Sid(accountSid), data, geolocationType);
+
+        if (geolocation.getResponseStatus() != null
```

Comment:
@gvagenas, it turned out that part of the code was correct, so after applying your suggestion and noticing it didn't work (sorry for rushing to commit here), I roll-backed commit dd5db33 in commit 866323c.

Part of the explanation is here: http://documentation.telestax.com/connect/api/geolocation-api.html#example-7-rejected-immediate-geolocation-request. The API needs to check if the result of validations turn out into a "rejected" kind of response, and provide feedback to the requester. That's why the code here looks a little complicated and redundant, but it needs to cover this logic as otherwise it won't be compliant with the API's definition and it breaks (we get an HTTP 404 error).